cm_sdk/models/__init__.py

```diff
@@ -0,0 +1,110 @@
+import json
+from json import JSONEncoder
+from enum import Enum
+
+__author__ = 'e0d'
+
+
+class CloudManagerJSONEncoder(JSONEncoder):
+    """
+    A custom JSON encoder to provide transparent conversion between
+    CloudManager JSON format and the Python classes that represent API
+    objects in the SDK.
+    """
+    # credit
+    # http://stackoverflow.com/questions/19053707/convert-snake-case-snake-case-to-lower-camel-case-lowercamelcase-in-python
+    def to_camel_case(self, snake_str):
+        # Deal with the case were @property creates class property 
+        # keys with an initial '_'
+        snake_str = snake_str.lstrip('_')
+        # Split remaining name on '_'
+        components = snake_str.split('_')
+        return components[0].lower() + "".join(x.title() for x in components[1:])
+
+    def default(self, o):  # pylint: disable=method-hidden
+        if isinstance(o, CloudManagerEnum):
+            return o.to_json()
+        elif hasattr(o, "__dict__"):
+            # converts class parameter names from snake case to camel case expected
+            # by the API and prunes nulls
+            return dict((self.to_camel_case(k), v) for k, v in o.__dict__.iteritems() if v)
+        else:
+            return None
+
+
+
+class CommonEqualityMixin(object):
+    """
+    Mixin class to provide deep eqality as equivalence for API objects 
+    """
+    def __eq__(self, other):
+
+        #
+        # must be an instances of the same class
+        #
+        if not isinstance(other, self.__class__):
+            return False
+
+        # expectation that keys are the same breaks for entities
+        # returned from the API
+        my_keys, other_keys = (sorted(self.__dict__.keys()), sorted(other.__dict__.keys()))
+
+        if not my_keys == other_keys:
+            return False
+
+        for k in my_keys:
+
+            v0 = self.__dict__[k]
+            v1 = other.__dict__[k]
```

Comment:
`__dict__` seems like the wrong way of going about this. That'll check every property Python baked into the class, not just the ones you care about. Perhaps it'd be better to store the "important" properties in a dict and just check that.

Actually, then you can just do `dict1 == dict2`.